components/Ethdriver/src/ethdriver.c

```diff
@@ -114,6 +114,18 @@ bool client_has_mac(unsigned int client_id);
 void client_get_mac(unsigned int client_id, uint8_t *mac);
 
 
+static client_t *eth_get_client(void)
+{
+    int id = client_get_sender_id();
+    client_t *client = NULL;
+    for (int i = 0; i < num_clients; i++) {
+        if (clients[i].client_id == id) {
+            client = &clients[i];
```

Comment:
This keeps the original implementation that iterates over all clients an returns the last match then. We could break the loop on the first match, assuming there is only one client with that ID. 
```suggestion
            return &clients[i];
```
Or put another error catcher here that warns if there are multiple matches.